test/ttmlir/Dialect/TTNN/sharding_matmul_0.mlir

```diff
@@ -0,0 +1,14 @@
+// RUN: ttmlir-opt --ttir-to-ttnn-backend-pipeline="enable-optimizer=true sharding-pass-enabled=true max-legal-layouts=0" %s | FileCheck %s
```

Comment:
The name of the test should try to explain what is being tested. I'd go with something like `max_legal_layouts_override_0.mlir`